Can't load uploaded CA certificate into the TrustManager on 0.42

**Describe the bug**
Whenever I try to use an uploaded CA certificate to establish a connection to a Postgres database, it fails with the following error:
```
Loading the SSL root certificate /var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_11006948831689911650.tmp into a TrustManager failed.
```

**Logs**
Server logs:
```
2022-02-08 14:09:25,565 ERROR driver.util :: Database connection error
org.postgresql.util.PSQLException: Loading the SSL root certificate /var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_9926679908450381105.tmp into a TrustManager failed.
	at org.postgresql.ssl.LibPQFactory.<init>(LibPQFactory.java:175)
	at org.postgresql.core.SocketFactoryFactory.getSslSocketFactory(SocketFactoryFactory.java:61)
	at org.postgresql.ssl.MakeSSL.convert(MakeSSL.java:34)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:534)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:149)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:213)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:51)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:223)
	at org.postgresql.Driver.makeConnection(Driver.java:465)
	at org.postgresql.Driver.connect(Driver.java:264)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:189)
	at clojure.java.jdbc$get_driver_connection.invokeStatic(jdbc.clj:271)
	at clojure.java.jdbc$get_driver_connection.invoke(jdbc.clj:250)
	at clojure.java.jdbc$get_connection.invokeStatic(jdbc.clj:411)
	at clojure.java.jdbc$get_connection.invoke(jdbc.clj:274)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1111)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invokeStatic(connection.clj:245)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invoke(connection.clj:242)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invokeStatic(connection.clj:253)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invoke(connection.clj:249)
	at metabase.driver.sql_jdbc$fn__81583.invokeStatic(sql_jdbc.clj:35)
	at metabase.driver.sql_jdbc$fn__81583.invoke(sql_jdbc.clj:33)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__19264.invoke(util.clj:48)
	at metabase.util$do_with_timeout$fn__6209.invoke(util.clj:352)
	at clojure.core$binding_conveyor_fn$fn__5772.invoke(core.clj:2034)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.security.cert.CertificateException: No certificate data found
	at java.base/sun.security.provider.X509Factory.parseX509orPKCS7Cert(X509Factory.java:461)
	at java.base/sun.security.provider.X509Factory.engineGenerateCertificates(X509Factory.java:361)
	at java.base/java.security.cert.CertificateFactory.generateCertificates(CertificateFactory.java:478)
	at org.postgresql.ssl.LibPQFactory.<init>(LibPQFactory.java:163)
	... 36 more
2022-02-08 14:09:25,567 ERROR api.database :: Cannot connect to Database
java.lang.Exception: Loading the SSL root certificate /var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_9926679908450381105.tmp into a TrustManager failed.
	at metabase.driver.util$can_connect_with_details_QMARK_.invokeStatic(util.clj:53)
	at metabase.driver.util$can_connect_with_details_QMARK_.doInvoke(util.clj:36)
	at clojure.lang.RestFn.invoke(RestFn.java:442)
	at metabase.api.database$test_database_connection.invokeStatic(database.clj:425)
	at metabase.api.database$test_database_connection.doInvoke(database.clj:415)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at metabase.api.database$fn__72963$test_connection_details__72968$fn__72969.invoke(database.clj:464)
	at metabase.api.database$fn__72963$test_connection_details__72968.invoke(database.clj:454)
	at metabase.api.database$fn__72995.invokeStatic(database.clj:492)
	at metabase.api.database$fn__72995.invoke(database.clj:478)
	at compojure.core$wrap_response$fn__32307.invoke(core.clj:160)
	at compojure.core$wrap_route_middleware$fn__32291.invoke(core.clj:132)
	at compojure.core$wrap_route_info$fn__32296.invoke(core.clj:139)
	at compojure.core$wrap_route_matches$fn__32300.invoke(core.clj:151)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319.invoke(core.clj:200)
	at metabase.server.middleware.auth$enforce_authentication$fn__65044.invoke(auth.clj:14)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319.invoke(core.clj:200)
	at compojure.core$make_context$handler__32347.invoke(core.clj:289)
	at compojure.core$make_context$fn__32351.invoke(core.clj:299)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$make_context$fn__32351.invoke(core.clj:300)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$make_context$fn__32351.invoke(core.clj:300)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$make_context$fn__32351.invoke(core.clj:300)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$make_context$fn__32351.invoke(core.clj:300)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$make_context$fn__32351.invoke(core.clj:300)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$make_context$fn__32351.invoke(core.clj:300)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at metabase.api.routes$fn__76088$fn__76091.invoke(routes.clj:56)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319.invoke(core.clj:200)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.server.routes$fn__76233$fn__76234.doInvoke(routes.clj:57)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319.invoke(core.clj:200)
	at compojure.core$make_context$handler__32347.invoke(core.clj:289)
	at compojure.core$make_context$fn__32351.invoke(core.clj:299)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32300.invoke(core.clj:153)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32300.invoke(core.clj:153)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__32300.invoke(core.clj:153)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319$f__32320$respond_SINGLEQUOTE___32321.invoke(core.clj:197)
	at metabase.server.routes$fn__76221$fn__76223.invoke(routes.clj:41)
	at compojure.core$routes$fn__32319$f__32320.invoke(core.clj:198)
	at compojure.core$routes$fn__32319.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__73255.invoke(exceptions.clj:98)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__73252.invoke(exceptions.clj:86)
	at metabase.server.middleware.log$log_api_call$fn__76699$fn__76700$fn__76701.invoke(log.clj:209)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invokeStatic(diagnostic.clj:15)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invoke(diagnostic.clj:9)
	at metabase.server.middleware.log$log_api_call$fn__76699$fn__76700.invoke(log.clj:201)
	at toucan.db$_do_with_call_counting.invokeStatic(db.clj:216)
	at toucan.db$_do_with_call_counting.invoke(db.clj:209)
	at metabase.server.middleware.log$log_api_call$fn__76699.invoke(log.clj:200)
	at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__80049.invoke(browser_cookie.clj:38)
	at metabase.server.middleware.security$add_security_headers$fn__54907.invoke(security.clj:148)
	at metabase.server.middleware.json$wrap_json_body$fn__79205.invoke(json.clj:62)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__79223.invoke(json.clj:98)
	at metabase.server.middleware.offset_paging$handle_paging$fn__54931.invoke(offset_paging.clj:42)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__80316.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__80332.invoke(params.clj:69)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__34400.invoke(misc.clj:59)
	at metabase.server.middleware.session$bind_current_user$fn__45736$fn__45737.invoke(session.clj:262)
	at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:243)
	at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:235)
	at metabase.server.middleware.session$bind_current_user$fn__45736.invoke(session.clj:261)
	at metabase.server.middleware.session$wrap_current_user_info$fn__45723.invoke(session.clj:221)
	at metabase.server.middleware.session$wrap_session_id$fn__45709.invoke(session.clj:167)
	at metabase.server.middleware.auth$wrap_api_key$fn__65052.invoke(auth.clj:27)
	at ring.middleware.cookies$wrap_cookies$fn__80236.invoke(cookies.clj:216)
	at metabase.server.middleware.misc$add_content_type$fn__34383.invoke(misc.clj:27)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__34408.invoke(misc.clj:76)
	at ring.middleware.gzip$wrap_gzip$fn__80278.invoke(gzip.clj:86)
	at metabase.server.middleware.misc$bind_request$fn__34411.invoke(misc.clj:93)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__80065.invoke(ssl.clj:38)
	at metabase.server$async_proxy_handler$fn__76472.invoke(server.clj:73)
	at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:386)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.postgresql.util.PSQLException: Loading the SSL root certificate /var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_9926679908450381105.tmp into a TrustManager failed.
	at org.postgresql.ssl.LibPQFactory.<init>(LibPQFactory.java:175)
	at org.postgresql.core.SocketFactoryFactory.getSslSocketFactory(SocketFactoryFactory.java:61)
	at org.postgresql.ssl.MakeSSL.convert(MakeSSL.java:34)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:534)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:149)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:213)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:51)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:223)
	at org.postgresql.Driver.makeConnection(Driver.java:465)
	at org.postgresql.Driver.connect(Driver.java:264)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:189)
	at clojure.java.jdbc$get_driver_connection.invokeStatic(jdbc.clj:271)
	at clojure.java.jdbc$get_driver_connection.invoke(jdbc.clj:250)
	at clojure.java.jdbc$get_connection.invokeStatic(jdbc.clj:411)
	at clojure.java.jdbc$get_connection.invoke(jdbc.clj:274)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1111)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invokeStatic(connection.clj:245)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invoke(connection.clj:242)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invokeStatic(connection.clj:253)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invoke(connection.clj:249)
	at metabase.driver.sql_jdbc$fn__81583.invokeStatic(sql_jdbc.clj:35)
	at metabase.driver.sql_jdbc$fn__81583.invoke(sql_jdbc.clj:33)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__19264.invoke(util.clj:48)
	at metabase.util$do_with_timeout$fn__6209.invoke(util.clj:352)
	at clojure.core$binding_conveyor_fn$fn__5772.invoke(core.clj:2034)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	... 1 more
Caused by: java.security.cert.CertificateException: No certificate data found
	at java.base/sun.security.provider.X509Factory.parseX509orPKCS7Cert(X509Factory.java:461)
	at java.base/sun.security.provider.X509Factory.engineGenerateCertificates(X509Factory.java:361)
	at java.base/java.security.cert.CertificateFactory.generateCertificates(CertificateFactory.java:478)
	at org.postgresql.ssl.LibPQFactory.<init>(LibPQFactory.java:163)
	... 36 more
2022-02-08 14:09:25,597 DEBUG middleware.log :: POST /api/database 400 281,3 ms (0 DB calls)
{:valid false,
 :dbname
 "Loading the SSL root certificate /var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_9926679908450381105.tmp into a TrustManager failed.",
 :message
 "Loading the SSL root certificate /var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_9926679908450381105.tmp into a TrustManager failed."}
```

**To Reproduce**
Steps to reproduce the behavior:
1. Start Metabase on 0.42-rc2 using the prebuilt JAR
2. Go to the "Add Database" admin page (/admin/databases/create)
3. Fill regular (host/port/username) connection details for a Postgres database using TLS
4. Turn "Use a secure connection (SSL)" on
5. Set SSL Mode to `verify-ca`
6. Upload a root CA X509 certificate
7. Press "Save"

**Expected behavior**
The connection with Postgres would have been successfully established.

**Information about your Metabase Installation:**

```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.110 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.12+0",
    "java.vendor": "Homebrew",
    "java.vendor.url": "https://github.com/Homebrew/homebrew-core/issues",
    "java.version": "11.0.12",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.12+0",
    "os.name": "Mac OS X",
    "os.version": "11.5.2",
    "user.language": "en",
    "user.timezone": "Europe/Berlin"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "1.4.197 (2018-03-18)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "1.4.197 (2018-03-18)"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2022-02-03",
      "tag": "v0.42.0-rc2",
      "branch": "release-x.42.x",
      "hash": "e119430"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
This is blocking my usage of Metabase Cloud, since I can't connect to my secured Postgres database: when self-hosting Metabase, I was already able to preinstall the root CA and my client certificates and pass them via the additional JDBC connection string, but I want to migrate to Metabase Cloud, which would require uploading the client certificates. I was testing 0.42-rc2 to see if it would fix my issue once it is rolled out to Metabase Cloud, after #19984 was fixed.

**Additional context**
There is a similar issue that happens when trying to pass the client certificate, but it fails to be loaded into the `KeyManager` instead of the `TrustManager`. You can only trigger that if you don't set the SSL mode to `certify-ca` or `certify-full`, as the CA cert error triggers first.

The file in `/var/folders/nx/ch0_9wmx7jq0tj65w85lts6w0000gn/T/metabase-secret_11006948831689911650.tmp` looks like:
```
data:application/x-x509-ca-cert;base64,LS0tLS1CRUdJTiBDRVJUSUZJQ0FU ...
```
where the last field is the uploaded cert base64-encoded.

--- a/src/metabase/models/secret.clj
+++ b/src/metabase/models/secret.clj
@@ -1,6 +1,7 @@
 (ns metabase.models.secret
   (:require [cheshire.generate :refer [add-encoder encode-map]]
             [clojure.java.io :as io]
+            [clojure.string :as str]
             [clojure.tools.logging :as log]
             [java-time :as t]
             [metabase.api.common :as api]
@@ -99,6 +100,8 @@
           (.write out v)))
       tmp-file)))
 
+(def ^:private uploaded-base-64-prefix "data:application/x-x509-ca-cert;base64,")
+
 (defn db-details-prop->secret-map
   "Returns a map containing `:value` and `:source` for the given `conn-prop-nm`. `conn-prop-nm` is expected to be the
   name of a connection property having `:type` `:secret`, and the relevant sub-properties (ex: -value, -path, etc.) will
@@ -121,27 +124,37 @@
         value-kw   (sub-prop "-value")
         options-kw (sub-prop "-options")
         id-kw      (sub-prop "-id")
-        value      (if-let [v (value-kw details)]     ; the -value suffix was specified; use that
-                     v
-                     (if-let [path (path-kw details)] ; the -path suffix was specified; this is actually a :file-path
-                       (do
-                         (when (premium-features/is-hosted?)
-                           (throw (ex-info
-                                    (tru "{0} (a local file path) cannot be used in Metabase hosted environment" path-kw)
-                                    {:invalid-db-details-entry (select-keys details [path-kw])})))
-                         path)
-                       (when-let [id (id-kw details)]
-                         (:value (Secret id)))))
-        source     (cond
-                     ;; set the :source due to the -path suffix (see above))
-                     (and (not= "uploaded" (options-kw details)) (path-kw details))
-                     :file-path
+        value      (cond
+                     ;; ssl-root-certs will need their prefix removed, and to be base 64 decoded (#20319)
+                     (and (value-kw details) (= "ssl-root-cert" conn-prop-nm)
+                          (str/starts-with? (value-kw details) uploaded-base-64-prefix))
+                     (-> (value-kw details) (str/replace-first uploaded-base-64-prefix "") u/decode-base64)
+
+                     ;; the -value suffix was specified; use that
+                     (value-kw details)
+                     (value-kw details)
+
+                     ;; the -path suffix was specified; this is actually a :file-path
+                     (path-kw details)
+                     (u/prog1 (path-kw details)
+                       (when (premium-features/is-hosted?)
+                         (throw (ex-info
+                                 (tru "{0} (a local file path) cannot be used in Metabase hosted environment" path-kw)
+                                 {:invalid-db-details-entry (select-keys details [path-kw])}))))
 
                      (id-kw details)
-                     (:source (Secret (id-kw details))))]
+                     (:value (Secret (id-kw details))))
+        source (cond
+                 ;; set the :source due to the -path suffix (see above))
+                 (and (not= "uploaded" (options-kw details)) (path-kw details))
+                 :file-path
+
+                 (id-kw details)
+                 (:source (Secret (id-kw details))))]
     (cond-> {:connection-property-name conn-prop-nm, :subprops [path-kw value-kw id-kw]}
       value
-      (assoc :value value, :source source))))
+      (assoc :value value
+             :source source))))
 
 (def
   ^{:doc "The attributes of a secret which, if changed, will result in a version bump" :private true}

--- a/test/metabase/models/secret_test.clj
+++ b/test/metabase/models/secret_test.clj
@@ -3,6 +3,7 @@
             [clojure.test :refer :all]
             [metabase.models.secret :as secret :refer [Secret]]
             [metabase.test :as mt]
+            [metabase.util :as u]
             [metabase.util.encryption-test :as encryption-test])
   (:import [java.io DataInputStream File]
            java.nio.charset.StandardCharsets))
@@ -73,3 +74,22 @@
                                     (with-open [in (DataInputStream. (io/input-stream val-file))]
                                       (.readFully in result))
                                     result))))))))))
+
+(deftest ssl-root-cert-base
+  (testing "db-details-prop->secret-map"
+    (testing "decodes root cert value properly (#20319)"
+      (is (= "<Certificate text goes here>"
+             (:value (secret/db-details-prop->secret-map
+                      {:ssl true
+                       :ssl-mode "verify-ca"
+                       :ssl-root-cert-value (str "data:application/x-x509-ca-cert;base64,"
+                                                 (u/encode-base64 "<Certificate text goes here>"))
+                       :ssl-root-cert-options "uploaded"
+                       :port 5432,
+                       :advanced-options false
+                       :dbname "the-bean-base"
+                       :host "localhost"
+                       :tunnel-enabled false
+                       :engine :postgres
+                       :user "human-bean"}
+                      "ssl-root-cert")))))))

